src/coffee/stockimport.coffee

```diff
@@ -130,27 +129,26 @@ class StockImport
 
   performCSV: (fileContent, next) ->
     new Promise (resolve, reject) =>
-      Csv().from.string(fileContent, {delimiter: @csvDelimiter, trim: true})
-      .to.array (data, count) =>
+      csv.parse fileContent, {delimiter: @csvDelimiter, trim: true}, (error, data) =>
+        if (error)
+          reject "#{CONS.LOG_PREFIX}Problem in parsing CSV: #{error}"
+
         headers = data[0]
         @_getHeaderIndexes headers, @csvHeaders
-        .then (mappedHeaderIndexes) =>
-          stocks = @_mapStockFromCSV _.tail(data), mappedHeaderIndexes[0], mappedHeaderIndexes[1]
-          debug "Stock mapped from csv for headers #{mappedHeaderIndexes}: %j", stocks
+          .then (mappedHeaderIndexes) =>
+            @_mapStockFromCSV(_.rest(data), headers).then (stocks) =>
+              debug "Stock mapped from csv for headers #{mappedHeaderIndexes}: %j", stocks
 
-          # TODO: ensure channel ??
-          @_perform stocks, next
-          .then (result) -> resolve result
-        .catch (err) -> reject err
-        .done()
-      .on 'error', (error) ->
-        reject "#{LOG_PREFIX}Problem in parsing CSV: #{error}"
+              @_perform stocks, next
+                .then (result) -> resolve result
+            .catch (err) -> reject err
+            .done()
 
   performStream: (chunk, cb) ->
     @_processBatches(chunk).then -> cb()
 
   _getHeaderIndexes: (headers, csvHeaders) ->
-    Promise.all _.map csvHeaders.split(','), (h) =>
+    Promise.all _.map csvHeaders.split(','), (h) ->
```

Comment:
Question: at this point, the parsed `csvHeaders` are always comma-separated or should we use the given `csvDelimiter` ?